src/python/pants/engine/http.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Dict, Optional, Tuple
+
+import requests
+
+from pants.engine.rules import side_effecting
+
+
+@dataclass(frozen=True)
+class HttpGetResponse:
+  url: str
+  status_code: Optional[int]
+  output_bytes: Optional[bytes]
+  headers: Tuple[Tuple[str, str], ...]
+
+
+@side_effecting
+class HttpRequester:
+
+  def get_request(self, url: str, headers: Dict[str, str] = {}) -> HttpGetResponse:
+    r = requests.get(url, headers)
+    header_list = [ (str(item), r.headers[item]) for item in r.headers ]
```

Comment:
Formatting nit:
```suggestion
    header_list = [(str(item), r.headers[item]) for item in r.headers]
```

(Oh how I'm excited for Black..)